### ISSUE TITLE
Wallet service loses wallet login

### ISSUE THREAD

**Issue by nebaughman** [blockchain/service-my-wallet-v3#63]
*Service v0.17.2, Node v0.12.10*

The wallet API service was started in the background via `nohup`, the `login` call succeeded, and wallet operations were working. At some point (days later), the API started generating errors that the wallet was not logged in. The service process was still running. Performing another login succeeded, and wallet operatoins worked again.

The `nohup.out` file contained the following (but it's not clear whether this was the cause of the problem):

```
...
ERR_WALLET_ID
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: listen EADDRINUSE
    at exports._errnoException (util.js:746:11)
    at Server._listen2 (net.js:1158:14)
    at listen (net.js:1184:10)
    at net.js:1282:9
    at dns.js:85:18
    at process._tickCallback (node.js:355:11)
ERR_WALLET_ID
...
```

**Comment by jtormey** [comment]
`EADDRINUSE` usually means node tried to connect to a port that is already in use. Not sure what the cause might be at this point, I'll look into it.

**Comment by nebaughman** [comment]
**Additional info:** In the case above, I started the service manually (not via System V init script, or such). I'm not using any keep-alive check yet. So, no automated script should have redundantly called the service.

**Caveat:** I haven't seen this happen again, so I can't be sure this error correlates with the failure case above. It is possible that I caused the `EADDRINUSE` by running the start script redundantly at some point when I was setting it up.

**Thought:** When telling the service to `start`, it seems to silently fail if another instance is already running. Consider printing a message (and exiting with a non-zero exit code) if another instance is running (or if anything is already bound to the port). This would help automation.

Thanks for looking into it! I'm not (yet) well-versed in nodejs behavior.
